### DISCUSSION TITLE
How to use Silero VAD in iOS and how to convert the Onnx to CoreMLmodel

### DISCUSSION THREAD

**Discussion by Kamalesh7605** [snakers4/silero-vad#556] [Q&A] [answered]
Hi everyone, I'm working on integrating Voice Activity Detection (VAD) technology using the Silero model in my iOS app. I would appreciate any guidance on how to implement the Silero model in an iOS environment, including best practices and relevant resources. Additionally, I need assistance with converting the ONNX model to a Core ML model. Any help would be greatly appreciated!

**Comment by snakers4** [comment, marked as answer]
Hi,

> Silero model in an iOS environment

Probably using ONNX runtime is the easiest option.

> converting the ONNX model to a Core ML model

Typically, such things are what we do commercially for companies for a fee.

**Reply by Kamalesh7605** [reply]
Hi, I’m looking for guidance on how to integrate an ONNX model into an iOS app. If there is any official documentation or detailed resources available for iOS integration using ONNX Runtime, could you please share them with me? Thanks!

**Reply by luizlacerdam** [reply]
Hello @snakers4 , I'm looking for the same integration. Could you give any more details? It is study porpuses.
Ty!

> Hi,
> 
> > Silero model in an iOS environment
> 
> Probably using ONNX runtime is the easiest option.
> 
> > converting the ONNX model to a Core ML model
> 
> Typically, such things are what we do commercially for companies for a fee.

**Reply by lumpidu** [reply]
If you are testing the ONNX runtime on iOS, it would be interesting to know the runtime memory requirements. You can use Instruments within XCode to see, how these grow over time. You can also tune in a limited way how memory is freed and used inside the memory pool.

In my experience, the benefit of CoreML is foremost in reducing the runtime memory requirements, as parts of the code can run on the neural engine. If you want to let your model run e.g. in audio unit extensions, this is very important.

**Comment by rootux** [comment]
@snakers4 Hey. We would love to get that commercial company service from you.
How can I contact you?

**Comment by helloooideeeeea** [comment]
Hi everyone,

I'm developing an iOS app that uses ONNX Runtime along with sileroVAD for automatic audio trimming. After reading through this discussion, I thought I might be able to help out. Over the past few days, I've open-sourced an audio trimming library that I've been working on. The repository currently includes both the library and a sample app. Although the README and source code documentation are still in progress, I plan to update them soon.

I hope this can be useful!

Repository: https://github.com/helloooideeeeea/RealTimeCutVADLibrary/

**Reply by ben2858898gft** [reply]
Thank you this is extremely helpful!

**Reply by helloooideeeeea** [reply]
@ben2858898gft 

Thank you! I'm glad the library is helpful. Since this project is complete, your star would be a great motivation for my future open-source work!

**Reply by ben2858898gft** [reply]
Yes definitely!
